mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,262 @@
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
+    ```
+
+    Some Python packages are installed as extensions of modules. For example, `numpy`, `scipy` and `pandas` are 
+    part of the `SciPy-bundle` module. You can use
+
+    ```bash
+    $ module show <module_name>
+    ```
+
+    to check which extensions are included in a module, if any.
+
+
+### Deactivating a virtual environment
+
+When you are done using the virtual environment, you can deactivate it.
+To do that, run:
+
+```bash
+$ deactivate
+```
+
+## Combining virtual environments with centrally installed modules
+
+You can combine Python packages installed in a virtual environment with environment modules. 
+The following script uses PyTorch (which is available as a module)
+and Poutyne (which we assume is not centrally installed):
+
+```python title="pytorch_poutyne.py"
+import torch
+import poutyne
+
+...
+```
+
+We load a PyTorch package as a module and install Poutyne in a virtual environment:
+
+```bash
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+$ pip install Poutyne
+```
+
+While the virtual environment is activated, we can run the script without any issues:
+
+```bash
+$ python pytorch_poutyne.py
+```
+
+
+## Creating a virtual environment for a specific cluster
+
+To create a virtual environment for a specific cluster, you need to start an interactive shell on that cluster.
+Let's say you want to create a virtual environment on the `donphan` cluster.
+
+```bash
+$ module swap cluster/donphan
+$ qsub -I
+```
+
+After some time, a shell will be started on the `donphan` cluster. 
+You can now create a virtual environment as described in [the first section](#creating-a-venv).
+This virtual environment can be used by jobs running on the `donphan` cluster.
+
+
+## Example Python job
+
+This section will combine the concepts discussed in the previous sections to:
+
+1. Create a virtual environment on a specific cluster.
+2. Combine packages installed in the virtual environment with modules.
+3. Submit a job script that uses the virtual environment.
+
+The example script that we will run is the following:
+
+```python title="pytorch_poutyne.py"
+import torch
+import poutyne
+
+print(f"The version of PyTorch is: {torch.__version__}")
+print(f"The version of Poutyne is: {poutyne.__version__}")
+```
+
+First, we create a virtual environment on the `donphan` cluster:
+
+```bash
+$ module swap cluster/donphan
+$ qsub -I
+# Load module dependencies
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+# install virtual environment dependencies
+$ pip install Poutyne
+$ deactivate
+```
+
+We exit the interactive shell by pressing `CTRL+D` and create a job script that loads the PyTorch module, 
```

Comment:
use `exit` command instead of `Ctrl-D`

Reply:
Fixed in ['exit' instead of CTRL+D to exit the interactive shell](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/598ce4f5bfe0dcb3bc3b41e307ce93eaa40ae267)